functional-tests.sh

```diff
@@ -334,6 +334,20 @@ function test_put_object_with_storage_class_error()
     log_success "$start_time" "${FUNCNAME[0]}"
 }
 
+## Test mc cp command with metadata set to a json string
+function test_put_object_with_metadata()
+{
+    show "${FUNCNAME[0]}"
+
+    start_time=$(get_time)
+    object_name="mc-test-object-$RANDOM"
+    assert_failure "$start_time" "${FUNCNAME[0]}" mc_cmd cp --attr '{"key1":"value1","key1":"value1"}' "${FILE_1_MB}" "${SERVER_ALIAS}/${BUCKET_NAME}/${object_name}"
```

Comment:
This functional test is incomplete, we should also add `mc stat` to validate if the attribute is set.